packages/codegen-ui-react/lib/forms/react-form-renderer.ts

```diff
@@ -447,6 +448,17 @@ export abstract class ReactFormTemplateRenderer extends StudioTemplateRenderer<
           ['id', lowerCaseDataTypeName],
         ),
       );
+
+      // Build effects to grab nested models off target record for relationships
+      Object.entries(formMetadata.fieldConfigs).forEach(([key, value]) => {
+        if (value.relationship) {
+          statements.push(
+            addUseEffectWrapper(buildGetRelationshipModels(key, lowerCaseDataTypeNameRecord), [
+              lowerCaseDataTypeNameRecord,
+            ]),
+          );
+        }
+      });
```

Comment:
This works but I think it is not good when there are multiple `hasOne` relationship field. It will end up creating multiple `useEffect` hooks. I think there is a simple way to do the same is modify the above `buildUpdateDataStoreQuery`, passing `fieldConfigs` to it.

This is what it generates now
```
  React.useEffect(() => {
    const queryData = async () => {
      const record = id ? await DataStore.query(Member, id) : member;
      setMemberRecord(record);
    };
    queryData();
  }, [id, member]);
```

After adding your update it should be:
```
```
  React.useEffect(() => {
    const queryData = async () => {
      const record = id ? await DataStore.query(Member, id) : member;
      const team = record ? await record.team : undefined;
      setTeam(team);
      setMemberRecord(record);
    };
    queryData();
  }, [id, member]);
```
```

Reply:
I had it in a separate effect becuase there could be multiple, and I wasnt sure if having all of it in one effect would look messy. I went back and forth when implementing this, but. can change it to use the other effect